plugins/volumeFilter_test.go

```diff
@@ -0,0 +1,439 @@
+			wantOp:    nil,
+			wantError: fmt.Errorf("could not convert amount () to float: strconv.ParseFloat: parsing \"\": invalid syntax"),
+			wantTBB:   nil,
+		},
+		{
+			name:      "failure - invalid amount in op",
+			filter:    emptyFilter,
+			dailyOTB:  nil,
+			dailyTBB:  nil,
+			op:        &txnbuild.ManageSellOffer{Buying: txnbuild.NativeAsset{}, Selling: txnbuild.NativeAsset{}, Price: "0.0"},
+			wantOp:    nil,
+			wantError: fmt.Errorf("could not convert amount () to float: strconv.ParseFloat: parsing \"\": invalid syntax"),
+			wantTBB:   nil,
+		},
+		{
+			name:      "success - selling, no filter sell caps",
+			filter:    emptyFilter,
+			dailyOTB:  &VolumeFilterConfig{},
+			dailyTBB:  &VolumeFilterConfig{SellBaseAssetCapInBaseUnits: createFloatPtr(0.0), SellBaseAssetCapInQuoteUnits: createFloatPtr(0.0)},
+			op:        &txnbuild.ManageSellOffer{Buying: txnbuild.NativeAsset{}, Selling: txnbuild.NativeAsset{}, Price: "1.0", Amount: "100.0"},
+			wantOp:    &txnbuild.ManageSellOffer{Buying: txnbuild.NativeAsset{}, Selling: txnbuild.NativeAsset{}, Price: "1.0", Amount: "100.0"},
+			wantError: nil,
+			wantTBB:   &VolumeFilterConfig{SellBaseAssetCapInBaseUnits: createFloatPtr(100.0), SellBaseAssetCapInQuoteUnits: createFloatPtr(100.0)},
+		},
+		{
+			name: "success - selling, base units sell cap, don't keep selling base",
+			filter: &volumeFilter{
+				name:                   "volumeFilter",
+				configValue:            "",
+				baseAsset:              testNativeAsset,
+				quoteAsset:             testNativeAsset,
+				config:                 &VolumeFilterConfig{SellBaseAssetCapInBaseUnits: createFloatPtr(0.0)},
+				dailyVolumeByDateQuery: dailyVolumeByDateQuery,
+			},
+			dailyOTB: &VolumeFilterConfig{
+				SellBaseAssetCapInBaseUnits:  createFloatPtr(0.0),
+				SellBaseAssetCapInQuoteUnits: createFloatPtr(0.0),
+			},
+			dailyTBB: &VolumeFilterConfig{
+				SellBaseAssetCapInBaseUnits:  createFloatPtr(0.0),
+				SellBaseAssetCapInQuoteUnits: createFloatPtr(0.0),
+			},
+			op:        &txnbuild.ManageSellOffer{Buying: txnbuild.NativeAsset{}, Selling: txnbuild.NativeAsset{}, Price: "1.0", Amount: "100.0"},
+			wantOp:    nil,
+			wantError: nil,
+			wantTBB:   &VolumeFilterConfig{SellBaseAssetCapInBaseUnits: createFloatPtr(0.0), SellBaseAssetCapInQuoteUnits: createFloatPtr(0.0)},
+		},
+		{
+			name: "success - selling, base units sell cap, keep selling base",
+			filter: &volumeFilter{
+				name:                   "volumeFilter",
+				configValue:            "",
+				baseAsset:              testNativeAsset,
+				quoteAsset:             testNativeAsset,
+				config:                 &VolumeFilterConfig{SellBaseAssetCapInBaseUnits: createFloatPtr(1.0), mode: volumeFilterModeExact},
+				dailyVolumeByDateQuery: dailyVolumeByDateQuery,
+			},
+			dailyOTB: &VolumeFilterConfig{
+				SellBaseAssetCapInBaseUnits:  createFloatPtr(0.0),
+				SellBaseAssetCapInQuoteUnits: createFloatPtr(0.0),
```

Comment:
I haven't checked the values on these tests yet but once we have a simpler interface to these test cases it will be a lot easier to go through. we should end up without any struct constructions in tests cases wherever possible.